### DISCUSSION TITLE
avoid overwriting file credentials

### DISCUSSION THREAD

**Discussion by chaoui** [theurichde/go-aws-sso#14] [Ideas]
Using the go-aws-sso tool, the credentials aws file is being overwritten with the new default profile and other attributes. The behavior we expected was to add the new information, keeping the old ones from other profiles.

Regards,
Daniel

**Comment by theurichde** [comment]
I am just thinking on how to approach that issue. The questions I have in mind are:
* How should one set the current to-be profile?
* How to deal with already existing profiles? 
  * Just override it?

**Reply by bfanyuk** [reply]
Hello, @theurichde 
Let me try to answer your questions.

> How should one set the current to-be profile

credentials file may contain several profiles
When user will execute specific command he/she will specify a profile to use with **--profile** option.
Both **go-aws-sso** and standard aws cli have this option.

> How to deal with already existing profiles

If user requested to login as existing profile, then we have no other choice than rewriting it.
Old **aws-cli-slc** does the same.

Please let me know in case of any questions.
Thanks!

**Comment by chaoui** [comment]
Hello Tim,

the issue is to avoid override actually. If the user chooses a different profile keep the old and new one... like in this image:

[cid:89b1641c-0b62-4d08-9edd-9885967077ab]
________________________________

**Reply by theurichde** [reply]
@chaoui Can you please re-upload the image? It seems that the image is broken

But I guess I get what you mean :-)

**Comment by fishgrimsby** [comment]
Hey @theurichde,

Great utility, thanks for the good work! I have made an attempt to fix the issue mentioned in this discussion and would welcome you taking a look 😄

#95 

Richard

**Reply by theurichde** [reply]
Hi @fishgrimsby! Thanks for your contribution ❤️  - I will take a look 👀
